src/sentry_core.c

```diff
@@ -934,6 +934,8 @@ sentry_transaction_finish(sentry_transaction_t *opaque_tx)
     sentry_value_t trace_context
         = sentry__value_get_trace_context(opaque_tx->inner);
     sentry_value_t contexts = sentry_value_new_object();
+    sentry_value_set_by_key(
+        contexts, "data", sentry_value_get_by_key(tx, "extra"));
```

Comment:
By the way, when you do a `sentry_value_set_by_key()`, where another ref owns the set value, you need to increase the ref count because `sentry_value_set_by_key()` doesn't do this (it is a ref sink, for convenience).

This is probably why you see ASAN and Valgrind cry: you add it to `contexts.trace`` here but later remove it from `tx`, at this point, it gets freed because there is still only one ref counted. Later, when we create the envelope and need to serialize it, we use that value after freeing it.